src/gpu/nvidia/cudnn_matmul_executor.hpp

```diff
@@ -380,35 +417,40 @@ struct cudnn_matmul_lt_runtime_args_exec_t final
     : public cudnn_matmul_lt_base_exec_t {
     status_t execute(const exec_ctx_t &ctx, impl::engine_t *engine,
             const std::shared_ptr<cudnn_matmul_lt_impl_t> matmul_impl_,
-            std::size_t algo_scratch_size, std::size_t bias_scratch_size,
-            std::size_t block_a_scratch_size, std::size_t block_b_scratch_size,
-            std::size_t block_c_scratch_size,
-            std::size_t src_scale_scratchpad_size,
-            std::size_t wei_scale_scratchpad_size) {
+            std::shared_ptr<cublas_lt_params> params,
+            const memory_desc_wrapper &src_d,
+            const memory_desc_wrapper &weights_d,
+            const memory_desc_wrapper &dst_d) {
+
+        cublas_lt_params matmul_params = *params;
```

Comment:
How big are the params structures? Given that we have to copy the whole structure during each execution we may want be cautions about their size. if it's heavy then we could split it into two parts: creation time and runtime and recreate only the runtime part in the case of runtime dimensions.

If the structures are not big and we don't anticipate them to grow in the future then I guess it's fine to copy the whole structure.

Reply:
There is only so much we can reduce the copy, a small refactor(another one) ensures we only allocate a new local params pointer for runtime dims cases and in any other we pass a shared_ptr.